### DISCUSSION TITLE
允许删除仅有一条关联的关联组

### DISCUSSION THREAD

**Discussion by DangoRepo** [kongying-tavern/map_register_v3#131] [Ideas]
在现有的后台，由于保存点位关联时关联数据不得为空，故若一个关联组内仅存在一条关联关系（为方便描述，以下用 `独体关联组` 代指这样的关联组）时，便无法通过删除关联线段的方式删除此条点位关联。
虽然可以使用 `特定的崩裂组操作*` 来删除此条点位关联，但我认为该操作属于非法操作，不仅需要建立在“该地区除该独体关联组外，仍存在点位关联”的前提下，且操作有相当程度不便，不应成为正规的删除关联的方式。

> 特定的崩裂组操作：先将需要删除的独体关联组与另一关联组进行崩裂组操作，再使用合并组重新正确连接关联组，从而达成将错误的独体关联组移除的目的。

建议：保存点位关联时，允许关联数据为空，并进行“数据是否为空”的检测；当该关联组下不存在关联数据时，则在数据层面上删除该关联组，防止由此操作产生过多的空关联组产生（我个人）无法预料的麻烦。

**Comment by Altermoe** [comment]
这个确实属于设计缺陷，我们目前正在研究怎么实现单条删除的功能。
